src/lib/worker-runner-protocol.js

```diff
@@ -85,6 +85,10 @@ class Protocol extends EventEmitter {
     this.transport.send(message);
   }
 
+  shutdown() {
+    this.send({type: 'shutdown', capabilities: [...this.capabilities]});
+  }
```

Comment:
This message type will require adding a  new capability to `supportedCapabilities`.

I'd prefer to have the protocol only implement `send`, and have the Host implementation call that.  Also, it should consult the protocol's capabilities and only send the message if it's supported.  I'm not sure what to do if it's not supported, though. Do we take that as an indication that this is not a system that should be shut down (static/standalone), in which case maybe `process.exit(0)` is the right choice?  Or that worker-runner doesn't support this functionality and docker-worker should do it directly, in which case invoking `shutdown -h now` is the right choice?

Reply:
> This message type will require adding a new capability to supportedCapabilities.

Ok

> I'd prefer to have the protocol only implement send, and have the Host implementation call that.

Makes sense

> Also, it should consult the protocol's capabilities and only send the message if it's supported. I'm not sure what to do if it's not supported, though

Do you mean sending a message to `worker-runner` querying the capabilities? I am thinking about when `worker-runner` receives a message type it doesn't support, it could reply with a message type like `unsupported-<message-type-received>`

Reply:
> Do you mean sending a message to worker-runner querying the capabilities? I am thinking about when worker-runner receives a message type it doesn't support, it could reply with a message type like unsupported-<message-type-received>

The protocol already includes a capability-negotiation phase, so you can just query `proto.capable('shutdown')`.